Presentation.Web/Infrastructure/Filters/AdvisSendFailureFilter.cs

```diff
@@ -0,0 +1,132 @@
+﻿using System;
+using System.Linq;
+using Core.ApplicationServices;
+using Core.DomainModel.Advice;
+using Core.DomainModel.Notification;
+using Core.DomainModel.Shared;
+using Core.DomainServices.Notifications;
+using Core.DomainServices.Repositories.Contract;
+using Core.DomainServices.Repositories.GDPR;
+using Core.DomainServices.Repositories.Project;
+using Core.DomainServices.Repositories.SystemUsage;
+using Hangfire.Common;
+using Hangfire.States;
+using Microsoft.Extensions.DependencyInjection;
+using Ninject;
+using Presentation;
+using Presentation.Web;
+using Presentation.Web.Infrastructure;
+using Presentation.Web.Infrastructure.Attributes;
+using Presentation.Web.Infrastructure.Filters;
+using Presentation.Web.Ninject;
+
+namespace Presentation.Web.Infrastructure.Filters
+{
+    public class AdvisSendFailureFilter : IElectStateFilter, IJobFilter
+    {
+        private static readonly string MatchType = typeof(AdviceService).FullName;
+        protected static readonly string MatchMethod = typeof(AdviceService)
+            .GetMethods()
+            .First(x => x.Name.Equals(nameof(AdviceService.SendAdvice)))
+            .Name;
+
+        private readonly StandardKernel _kernel;
+
+        public AdvisSendFailureFilter(StandardKernel kernel)
+        {
+            _kernel = kernel;
+        }
+
+        public void OnStateElection(ElectStateContext context)
+        {
+            if (context.CandidateState is FailedState failedState)
+            {
+                if(context.GetJobParameter<bool>("NoMoreRetries"))
+                {
+                    if (IsSendAdvice(context))
+                    {
+                        // Get argument input to method. In this case it's only one argument and it is the Id of the advice.
+                        int adviceId = Convert.ToInt32(context.BackgroundJob.Job.Args.First());
+                        if(adviceId > 0)
+                        {
+                            using (new HangfireNinjectResolutionScope(_kernel))
+                            {
+                                var advisService = _kernel.GetService<IAdviceService>();
```

Comment:
try catch rundt om det her og log fejlen hvis der kommer en for vi kan ikke leve med at fejl-filteret fejler :-)